lua/dap.lua

```diff
@@ -460,7 +460,11 @@ function Session:event_stopped(stopped)
       if not current_frame then
         return
       end
-      jump_to_frame(current_frame, stopped.preserveFocusHint)
+      local preserve_focus
+      if stopped.reason ~= 'pause' then
+        preserve_focus = stopped.preserveFocusHint
+      end
+      jump_to_frame(current_frame, preserve_focus)
```

Comment:
I always jump to frame on `pause`. My debug adapter sends `preserveFocusHint=true`. But if we only pause a single thread explicitly we probably want to jump to the respective frame.